src/porting.cpp

```diff
@@ -486,8 +491,11 @@ bool setSystemPaths()
 	}
 
 #ifndef __ANDROID__
-	path_user = std::string(getHomeOrFail()) + DIR_DELIM "."
-		+ PROJECT_NAME;
+	path_user = std::string(getenv("MINETEST_USER_PATH"));
```

Comment:
getenv returns null if unset, constructing a string from a null pointer is undefined behaviour